packages/lib-user/src/components/shared/MainContent/MainContent.js

```diff
@@ -67,139 +46,187 @@ function MainContent({
   source = DEFAULT_SOURCE
 }) {
   const [activeTab, setActiveTab] = useState(0)
+  const [showCalendar, setShowCalendar] = useState(false)
+  const [customDateRange, setCustomDateRange] = useState([selectedDateRange.startDate, selectedDateRange.endDate])
+  
   const handleActiveTab = useCallback((tabIndex) => {
     setActiveTab(tabIndex)
   }, [])
 
+  useEffect(function updateCustomDateRange() {
+    setCustomDateRange([selectedDateRange.startDate, selectedDateRange.endDate])
+  }, [selectedDateRange])
+  
   const size = useContext(ResponsiveContext)
 
   const hoursSpent = convertStatsSecondsToHours(stats?.time_spent)
 
-  // create project options
-  let projectOptions = [
-    { label: 'ALL PROJECTS', value: 'AllProjects' },
-    ...(projects || []).map(project => ({
-      label: project.display_name,
-      value: project.id
-    }))
-  ]
-  const selectedProjectOption = projectOptions.find(option => option.value === selectedProject)
-
-  // create date range options
-  const dateRangeOptions = getDateRangeSelectOptions(source?.created_at?.substring(0, 10))
+  const { dateRangeOptions, selectedDateRangeOption } = getDateRangeSelectOptions({ created_at: source?.created_at?.substring(0, 10), selectedDateRange })
```

Comment:
It's a better idea long-term to abstract out the `source?.created_at?.substring(0, 10)` into a named helper function that gets reused for this type of data/value extraction. I'd have something like, `getDateString()` or `getDateYYYY_MM_DD(source?.created_at)`. That way, when you have a validation method, you can put it in there and then pass in a 2nd prop for a default value if the date you're trying to validate is invalid.

Reply:
Great idea, added `getStatsDateString` accordingly 🚀